packages/extension/src/companion/useExtensionPermission.ts

```diff
@@ -77,16 +72,53 @@ export const useExtensionPermission = ({
       });
       client.setQueryData(contentScriptKey, true);
       await registerBrowserContentScripts();
-      window.open(companionPermissionGrantedLink, '_blank');
+
+      if (onPermission) {
+        await onPermission(true);
+      } else {
+        window.open(companionPermissionGrantedLink, '_blank');
+      }
+    } else {
+      trackEvent({
+        event_name: 'decline content scripts',
+        extra: JSON.stringify({ origin }),
+      });
+
+      if (onPermission) onPermission(false);
     }
 
     return granted;
   };
+};
+
+export const useContentScriptStatus = (): {
```

Comment:
I'm not really sure why we need another hook for this?
Don't we have this knowledge already somewhere (the one we base the companion button off?)

Reply:
it's not exactly another hook. It is the same function which previously was implemented inside the `useExtensionPermission` hook -- I extracted it out so we can dynamically attach it.

The reason you see duplication here is because the legacy `useExtensionPermission` hook is still used in a few places, so replacing it entirely would be out of scope for this task